### ISSUE TITLE
🖼️ Unreadable sponsorship logos (images are skewed)

### ISSUE THREAD

**Issue by jefftriplett** [django/djangoproject.com#1816]
Many of the Sponsorship logos are not readable. The logos are stretched and skewed. 

Semi-related to #1431 

![Screenshot 2024-12-10 at 8 24 34 AM](https://github.com/user-attachments/assets/3cb39ec6-60e8-4120-85ba-4dd4ef7886db)

**Comment by bmispelon** [comment]
Oh interesting, thanks for flagging this!

I wonder if this was caused by the removal of unveil.js in 6757a83ffc82a0710f5f3c1ee3fd22d3be98695b or if this bug was present before 🤔 

I'll ping @adamzap just in case

**Comment by adamzap** [comment]
@bmispelon I think this is happening because of the explicit width and height attributes you wanted in 6757a83f. Maybe we're using the wrong dimension attribute there?

**Comment by bmispelon** [comment]
Ah right, that's probably it. I used the hardcoded `THUMBNAIL_SIZE` to avoid reading the file on disk (which I think needs to happen in order to get its width/height, but maybe I'm wrong 🤔 ).

**Comment by bmispelon** [comment]
According to [sorl-thumbnail's documentation](https://sorl-thumbnail.readthedocs.io/en/latest/operation.html) (very last paragraph, emphasis mine):

> We can store meta data in the Key Value Store that would be too costly to retrieve even for local file storage. Today this meta data consists only of the **image size** but this could be expanded to for example EXIF data.

So it seems it should be safe to use the thumbnail's actual size instead of the `THUMBNAIL_SIZE` constant (a test for it would be nice though, I'll see if I can come up with something)

**Comment by sodevious** [comment]
Suggesting a CSS-only change that would help keep ratio of the images:
`max-height: fit-content;`

[Docs on fit-content](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content) suggest most modern browsers would support it.

Though previous suggestions of using the actual thumbnail size would probably be better for accessibility?

**Comment by bmispelon** [comment]
> Suggesting a CSS-only change that would help keep ratio of the images:
> `max-height: fit-content;`

I tried this on the [donation page](https://www.djangoproject.com/fundraising/#dsf-supporters) but it didn't seem to fix anything:
```
.heroes-section .heroes .hero-logo img {
  vertical-align: middle;
  max-height: fit-content;
}
```
Should the property be applied to the `img` or its parent container?

In any case, I've made a PR with the fix that Adam suggested in https://github.com/django/djangoproject.com/issues/1816#issuecomment-2532671566 and it appears to work (see screenshot on PR). I wasn't able to verify with a test that the thumbnail sizes are actually cached, but I guess Sentry will tell us if that's the case when we deploy to production.
